colibre/scripts/cosmic_log_SNIa_rate_CSFH_based.py

```diff
@@ -0,0 +1,307 @@
+"""
+Plots the star formation history. Modified version of the script in the
+github.com/swiftsim/swiftsimio-examples repository.
+"""
+import unyt
+
+import matplotlib.pyplot as plt
+import numpy as np
+import glob
+
+from swiftsimio import load
+
+from load_sfh_data import read_obs_data
+
+from velociraptor.observations import load_observations
+
+from astropy.cosmology import z_at_value
+from astropy.units import Gyr
+
+sfr_output_units = unyt.msun / (unyt.year * unyt.Mpc ** 3)
+log_multiplicative_factor = 4
+multiplicative_factor = 10 ** log_multiplicative_factor
+SNIa_output_units = 1. / (unyt.year * unyt.Mpc ** 3)
+
+from swiftpipeline.argumentparser import ScriptArgumentParser
+
+def power_law_beta_one_DTD(t, t_delay, tH):
+    mask = t > 0.04
+    value = np.zeros(len(t)) / unyt.Gyr 
+    value_new = 1./(np.log(tH) - np.log(t_delay)) / t[mask]
+    value_new.convert_to_units("Gyr**-1")
+    value[mask] = value_new 
+    return value
+
+def power_law_DTD(t, t_delay, tH, beta):
+    mask = t > 0.04
+    value = np.zeros(len(t)) / unyt.Gyr 
+    value_new = (1.-beta)/(tH**(1-beta) - (t_delay)**(1-beta)) / t[mask]**beta
+    value_new.convert_to_units("Gyr**-1")
+    value[mask] = value_new 
+    return value
+
+
+def exponential_law_DTD(t, t_decay, t_delay):
+    mask = t > 0.04
+    value = np.zeros(len(t)) / unyt.Gyr
+    value_new = np.exp(-(t[mask]-t_delay)/t_decay)/t_decay
+    value_new.convert_to_units("Gyr**-1")
+    value[mask] = value_new 
+    return value
+
+def Gaussian_law_DTD(t, t_delay, tmean, tsigma):
```

Comment:
Is this because the function uses the hard-cored value of 0.04? Is this correct or should it be `t_delay`?

If you change something here, note that this function is present in some other files too.